pkgs/stdenv/generic/make-derivation.nix

```
@@ -128,6 +128,10 @@ let
     else lib.subtractLists hardeningDisable (defaultHardeningFlags ++ hardeningEnable);
   # hardeningDisable additionally supports "all".
   erroneousHardeningFlags = lib.subtractLists supportedHardeningFlags (hardeningEnable ++ lib.remove "all" hardeningDisable);
+
+  checkDependencyList = name: deps: lib.flip lib.imap1 deps (index: dep:
+    if lib.isDerivation dep || isNull dep || builtins.typeOf dep == "path" then dep
+    else throw "Dependency is not a derivation: element ${toString index} of ${name} for ${attrs.name or attrs.pname}");
```

```suggestion
    else throw "Dependency is not a valid type: element ${toString index} of ${name} for ${attrs.name or attrs.pname}");
```
maybe, since null and paths are in fact supported?

---

Done :)